### PR TITLE
Extract RSpec Rails cops

### PR DESCRIPTION
**Pull request by ydah**
Fix: https://github.com/rubocop/rubocop-rspec/issues/1734

### TODO
- [x] once the gem is published, adds rubocop-rspec_rails to gemspec and remove from Gemfile
- [x] set the version constraint in gemspec for `rspec-rspec_rails`
- [x] Creating a Repository for https://github.com/rubocop/rubocop-rspec_rails
- [x] Extracted to https://github.com/rubocop/rubocop-rspec_rails

Here's where to temporary extract it:.
- https://github.com/ydah/rubocop-rspec_rails/pull/1

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [ ] Added tests.
- [ ] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Comment by pirj** [comment]
Should we do rubocop-rspec_rails with underscore lije we do for rubocop-factory_bot?
As a repo name and in the code especially: we’ll use RuboCop::RSpecRails as the namespace, right?

**Comment by bquorning** [comment]
Based on https://guides.rubygems.org/name-your-gem/ and e.g. https://github.com/ydah/rubocop-rspec-rails/blob/207fc3483e75605b12276b90e5bb3a7ebcdc223f/lib/rubocop/cop/rspec_rails/avoid_setup_hook.rb defining `RuboCop::Cop::RSpecRails` and not `RuboCop::Cop::RSpec::Rails`, I think the gem should be named rubocop-rspec_rails.

**Comment by ydah** [comment]
Thank you. I changed rubocop-rspec_rails instead of rubocop-rspec-rails.

**Comment by bquorning** [comment]
I created https://github.com/rubocop/rubocop-rspec_rails with @rubocop/rubocop-core given write-access. Ping me if anyone needs admin access.

**Comment by ydah** [comment]
If the modifications on the rubocop-rspec_rais side are correct, we will apply them to the repository you created. Please review https://github.com/ydah/rubocop-rspec_rails/pull/1 .

File: config/obsoletion.yml [hunk 1]
```diff
@@ -27,3 +27,10 @@ renamed:
   RSpec/FactoryBot/FactoryClassName: FactoryBot/FactoryClassName
   RSpec/FactoryBot/FactoryNameStyle: FactoryBot/FactoryNameStyle
   RSpec/FactoryBot/SyntaxMethods: FactoryBot/SyntaxMethods
+  RSpec/Rails/AvoidSetupHook: RSpecRails/AvoidSetupHook
```
Comment (pirj) [inline config/obsoletion.yml:30]: The rest should be this way, too

File: docs/modules/ROOT/pages/departments.adoc [hunk 2]
```diff
@@ -35,7 +35,7 @@ There are plans for extracting the following three departments before RuboCop RS
 |Extracted in https://github.com/rubocop/rubocop-rspec/releases/tag/v2.22.0[v2.22.0]
 
 |RSpec/Rails
-|rubocop-rspec-rails
+|rubocop-rspec_rails
 |Not yet extracted
```
Comment (pirj) [inline docs/modules/ROOT/pages/departments.adoc:39]: Extracted! 🥳
```suggestion
|Extracted in https://github.com/rubocop/rubocop-rspec/releases/tag/v2.28.0[v2.28.0]
```

File: lib/rubocop/cop/rspec/rails/http_status.rb [hunk 3]
```diff
@@ -1,213 +1,60 @@
 # frozen_string_literal: true
 
-require 'rack/utils'
```
Comment (pirj) [inline lib/rubocop/cop/rspec/rails/http_status.rb:3]: We can now remove rack-utils from the gemspec?

**Review by pirj — APPROVED** [review]
Looks great, thank you!
Were there any modules that are needed to to run the newly-extracted extension without rubocop-rspec? Can we remove them from rubocop-rspec? I recall we had that for capybara/factory_bot.

**Comment by pirj** [comment]
I can’t spot any rails-specific modules in our code at a glance, so all good.

File: lib/rubocop/cop/rspec/rails/http_status.rb (hunk 3 above)
Comment (ydah) [reply lib/rubocop/cop/rspec/rails/http_status.rb:3]: Yes! Thank you. I remove it.

File: config/obsoletion.yml (hunk 1 above)
Comment (ydah) [reply config/obsoletion.yml:30]: Ah, sorry. I fixed it.

**Review by pirj — APPROVED** [review]
Lgtm! 😍

**Comment by ydah** [comment]
I overlooked something serious. Gems cannot depend on each other.

```
❯ bundle install
Your bundle requires gems that depend on each other, creating an infinite loop. Please remove either gem 'rubocop-rspec' or gem 'rubocop-rspec_rails' and try again.
```

**Comment by pirj** [comment]
I guess we’ll have to remove the dependency from rubocop-rspec_rails and make a note there that it implicitly depends on rubocop-rspec.
In 3.0, we’ll remove the dependency on rubocop-rspec_rails from rubocop-rspec, and will restore the dependency declaration in rubocop-rspec_rails.
This might be a mild surprise to those who would want just rubocop-rspec_rails, but I suppose it’s a very rare case.

**Comment by ydah** [comment]
@pirj Should rubocop-rspec_rails be a patch upgrade or a minor upgrade?

**Comment by pirj** [comment]
I’d yank the 2.28.0 and released a bugfix patch in 2.28.1.
@bquorning what do you think?

**Comment by bquorning** [comment]
>I’d yank the 2.28.0 and released a bugfix patch in 2.28.1.
Yes that sounds like a good idea. And I assume nobody is using the gem yet, so the yank won’t affect users.

**Comment by ydah** [comment]
I update this PR. And All Green ✅

**Pull request merged**